src/components/App/App.js

```diff
@@ -1,5 +1,43 @@
-import { Frontpage } from 'pages';
+import { Switch, Route, Redirect } from 'react-router-dom';
+import { Frontpage, Dashboard } from 'pages';
+import { useState } from 'react';
 
 export const App = () => {
-    return <Frontpage />;
+    const [user, setUser] = useState(() => {
+        if (localStorage.getItem('user')) {
+            return JSON.parse(localStorage.getItem('user'));
+        } else return [];
+    });
+
+    const userIsAuthenticated = user.length !== 0;
+
+    const authenticateUser = array => {
+        localStorage.setItem('user', JSON.stringify(array));
+        setUser(array);
+    };
+
+    const logoutUser = () => {
+        localStorage.removeItem('user');
+        setUser([]);
+        document.title = 'Anmelden | Student Compass';
+    };
+
+    return (
+        <Switch>
+            <Route exact path="/">
+                {userIsAuthenticated ? (
+                    <Redirect to="/dashboard" />
+                ) : (
+                    <Frontpage authenticateUser={authenticateUser} />
```

Comment:
A function that you give down as a prop would be from the naming like:

```suggestion
                    <Frontpage onAuthenticateUser={authenticateUserHandler} />
```

The **on** is the important one here, so that someone else who sees the FrontPage component would know immediately that it is a function that it receives as a prop.

Same for the Dashboard in line 36